Source/SwiftLintFramework/Rules/Idiomatic/ConvenienceTypeRule.swift

```diff
@@ -87,11 +131,35 @@ public struct ConvenienceTypeRule: ASTRule, OptInRule, ConfigurationProviderRule
             return []
         }
 
-        return [
-            StyleViolation(ruleDescription: type(of: self).description,
-                           severity: configuration.severity,
-                           location: Location(file: file, byteOffset: offset))
-        ]
+        // MARK: Has @objc members?
+        let hasObjcMembers = dictionary.substructure.contains { dict in
+            return dict.enclosedSwiftAttributes.contains(.objc)
+        }
+
+        guard !hasObjcMembers else {
+            return []
+        }
+
+        let makeViolation = {
+            return [
+                StyleViolation(ruleDescription: type(of: self).description,
+                               severity: self.configuration.severity,
+                               location: Location(file: file, byteOffset: offset))
+            ]
+        }
+
+        // MARK: Is class?
+        guard kind == .class else {
+            return makeViolation()
+        }
+
+        // MARK: Is final?
+        let isFinal = dictionary.swiftAttributes.contains { attributes in
```

Comment:
since this is an opt-in rule, we might want to consider allowing some false positives (non final classes that actually are inherited from) vs increasing the number of false negatives (non final classes that are not inherited from) as I'd expect that in most cases those classes wouldn't be inherited from due to their nature

Reply:
Can we detect whether the class is inherited from inside the module? That, plus non-public, could be a nice thing to check.

Reply:
only with the "analyzer" mode, which requires a full clean build log

Reply:
In that case, would you mind giving examples of triggering or non-triggering examples that differ from the current PR? There are enough double-negatives in this conversation that I'm having trouble keeping it all straight 😅

Reply:
I think we should trigger in this case for example: https://github.com/wordpress-mobile/WordPress-iOS/blob/35e994e735b966ee2ef8bf0da52a9cb6f3e1bdec/WordPress/Classes/ViewRelated/Media/MediaNoticeNavigationCoordinator.swift#L6:1

Reply:
Makes sense to me. Thanks for taking the time to explain it. I pushed an update that addresses this.